prdoc/pr_9067.prdoc

```diff
@@ -0,0 +1,21 @@
+title: '[AHM] Derive DecodeWithMemTracking and pub fields'
+doc:
+- audience: Runtime Dev
+  description: |-
+    - Derive `DecodeWithMemTracking` on structs
+    - Make some fields public
+crates:
+- name: pallet-balances
+  bump: major
```

Comment:
None of the changes you have done require a major bump.

Reply:
CI said major... I can put it to minor again to demonstrate.

Reply:
You are just implementing a new trait and making fields public, not sure how this should influence the semver.